test/unit/deadline_client/cli/test_cli_asset.py

```diff
@@ -0,0 +1,171 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+from unittest.mock import patch, Mock
+from click.testing import CliRunner
+
+from deadline.client.cli import main
+from deadline.client import api
+from deadline.job_attachments.upload import S3AssetManager
+from deadline.job_attachments.models import AssetRootGroup
+
+
+class TestSnapshot:
+
+    def test_snapshot_root_directory_only(self, tmp_path):
+        """
+        Tests if CLI snapshot command calls correctly with the --root-dir arguement
+        """
+        root_dir = str(tmp_path)
+
+        temp_file = tmp_path / "temp_file.txt"
+        temp_file.touch()
+
+        with patch.object(
+            S3AssetManager, "prepare_paths_for_upload"
+        ) as prepare_paths_mock, patch.object(
+            api, "hash_attachments", return_value=(Mock(), [])
+        ) as hash_attachments_mock:
+
+            asset_group_mock = AssetRootGroup(
+                root_path=root_dir,
+                inputs=set(),
+                outputs=set(),
+                references=set(),
+            )
+            upload_group_mock = Mock(
+                asset_groups=[asset_group_mock],
+                total_input_files=1,
+                total_input_bytes=100,
+            )
+            prepare_paths_mock.return_value = upload_group_mock
+
+            runner = CliRunner()
+            result = runner.invoke(main, ["asset", "snapshot", "--root-dir", root_dir])
+
+            assert result.exit_code == 0
+            prepare_paths_mock.assert_called_once_with(
+                input_paths=[str(temp_file)], output_paths=[root_dir], referenced_paths=[]
+            )
+            hash_attachments_mock.assert_called_once()
```

Comment:
Do we still need the `patch.object` with blocks active while we are asserting the test results?